### ISSUE TITLE
[Feature Request] Solve In Place

### ISSUE THREAD

**Issue by iamrecursion** [LiamRiddell/obsidian-solve#43]
**Is your feature request related to a problem? Please describe.**
Recently I've found myself wanting to write an expression for solving in-line, and then evaluate it into place. 

**Describe the solution you'd like**
Imagine a dataview field as follows:

```md
**My Field**::  
[My other field:: ]
```

I would love to be able to write something like this (syntax by way of example, not this specifically):

```md
**My Field**:: s`1 + 1`
[My other field:: s`cos(pi)`]
```

Then I could either leave those to see the expression and result evaluated in place (which would not work with dataview but could be useful in other places):

```md
**My Field**:: s`1 + 1` = 2
[My other field:: s`cos(pi)` = -1]
```

Or I could commit the result. Upon committing the result, in this case the expression _would be removed_, just leaving the result.

```md
**My Field**:: 2
[My other field:: -1]
```

**Describe alternatives you've considered**
At the moment I just tend to write the calculation out of line, commit the result, and then copy that into place. 

**Additional context**
Nothing other than this is very much a nice to have, not something I would consider particularly high priority.

**Comment by iamrecursion** [comment]
@LiamRiddell is this something you're open to adding? I might have some time soon to give it a go if so.

**Comment by LiamRiddell** [comment]
@iamrecursion 👋 Sorry for the delayed response... my time is really restricted right now. I really like this idea and I feel it could be a great addition to Solve. I'm more than happy for you to give it a go and I can support you. You could claim the first collaborator medal!! 🎖️ 

Initially, I think the implementation steps would be:
1. In the pre-processor pipeline find the pattern s`SOLVE ME` in the input line and pass that onto the next stage.
NOTE: We may need to create a new pipeline class that supports providing a mutable context that can change as it passes through stages of the pipeline. This would to be able to store the original character offset of the expression and the length of the expression for step 3/4.
3. Displaying the result will require us to be able to pass an override index to the decoration widget for `character index + expression length` that the expression was found.
4. Updating the `Click to commit` event handler to replace the original expression.

If you come up with any better approach then I'm more than happy for you to go with it and we can review it!

**Comment by iamrecursion** [comment]
No worries at all! I'll have a poke at it soonish, assuming that life goes according to plan! Will keep you updated here.

**Comment by iamrecursion** [comment]
Goodness me I've had _no_ chance to look at this yet. Life has been unkind. I'll still try and look at it at some point, but please don't let me keep you from it if you ever have the time.

**Comment by LiamRiddell** [comment]
> Goodness me I've had _no_ chance to look at this yet. Life has been unkind. I'll still try and look at it at some point, but please don't let me keep you from it if you ever have the time.

Hello! 👋 Now that I've got a bit more time, I've picked up the above ticket. A new settings will be added to determine what the format of the substitution is when commiting the result:

![image](https://github.com/LiamRiddell/obsidian-solve/assets/3812154/a68a598a-29e5-41d0-a1a0-be367a7e9dca)

The solve will display inline at the location of the solve:
![image](https://github.com/LiamRiddell/obsidian-solve/assets/3812154/3d1f1064-585a-4ea9-a4be-87acfb2bb837)

The solve can be committed as usual:
![](https://i.gyazo.com/9b5e41385cf64a8af494bf8b799558c1.gif)

This feature will be available in version 1.0.0. 👍

**Comment by iamrecursion** [comment]
Oh this is _incredible_ I am going to get so much use out of this! Thank you _so_ much.

With regards to inserted formatting, does it always put it with the equals and in backticks? Most of my use-cases want to instead insert without the equals and just have the result directly as I plan to use this to compute values for dataview fields and the like.
